semgrep/semgrep/output.py

```diff
@@ -97,7 +101,26 @@ def finding_to_line(
                 else:
                     line_number = f"{start_line + i}"
 
+                if per_line_max_chars_limit and len(line) > per_line_max_chars_limit:
+                    stripped = True
+                    is_first_line = i == 0
+                    if is_first_line:
+                        line = (
+                            line[
+                                start_col - 1 : start_col - 1 + per_line_max_chars_limit  # type: ignore
+                            ]
+                            + ELLIPSIS_STRING
+                        )
+                        if start_col > 1:  # type: ignore
+                            line = ELLIPSIS_STRING + line
+                    else:
+                        line = line[:per_line_max_chars_limit] + ELLIPSIS_STRING
+                    line = line + colorama.Style.RESET_ALL
```

Comment:
```suggestion
```
Is this a left over from having the line be yellow in the screen shot?

Reply:
No, this was added to handle a case where a section is highlighted but trailing RESET_ALL is cut off. In which case following text will also get coloured untill another RESET_ALL is encountered. I am not sure if this is most graceful way to handle it.


I tried to handle over-all stripping of lines in two ways, before coloring and after coloring. Doing it before coloring is more complicated because of the way "coordinates" are passed to function that colours it.

Reply:
Now that I think about it, better way would be to add it in last matched line and not universally.

Sounds good?

Reply:
Hmm i see. Can you add a comment above this line mentioning that it is to handle that case

Reply:
Done @brendongo